File: javaapp/src/main/java/com/scandit/shelf/javaapp/ui/storeselection/StoreItemViewModel.java
package com.scandit.shelf.javaapp.ui.storeselection;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.scandit.shelf.sdk.catalog.Store;

/**
 * A ViewModel that exposes the currently selected Store to Activity/Fragment
 */
public class StoreItemViewModel {
    // Communicates with Fragment when Store selection in stores list changes.
    private final MutableLiveData<Store> selectedStoreLiveData;

    // Refers to the current Store item bound through StoresAdapter
    private Store item = null;

    MutableLiveData<String> _storeName = new MutableLiveData<>();
    // Observes Store item binding in StoresAdapter
    LiveData<String> storeName = _storeName;

    public StoreItemViewModel(MutableLiveData<Store> selectedStoreLiveData) {
        this.selectedStoreLiveData = selectedStoreLiveData;
    }

    public void bind(Store store) {
        item = store;
        _storeName.postValue(store.getName());
    }

    public void onClick() {
        if (item != null) {
            selectedStoreLiveData.postValue(item);
        }
    }
}

File: kotlinapp/src/main/kotlin/com/scandit/shelf/kotlinapp/ui/storeselection/StoreItemViewModel.kt
package com.scandit.shelf.kotlinapp.ui.storeselection

import androidx.lifecycle.MutableLiveData
import com.scandit.shelf.sdk.catalog.Store
import kotlinx.coroutines.channels.Channel

/**
 * A ViewModel that exposes the currently selected Store to Activity/Fragment.
 */
class StoreItemViewModel(
    // Communicates with Fragment when Store selection in stores list changes.
    private val storeChannel: Channel<Store>
) {
    // Refers to the current Store item bound through [StoresAdapter].
    var item: Store? = null

    // Observes Store item binding in StoresAdapter.
    val storeName = MutableLiveData<String>()

    fun bind(store: Store) {
        item = store
        storeName.postValue(store.name)
    }

    fun onClick() {
        item?.let { storeChannel.trySend(it) }
    }
}
